django/contrib/auth/hashers.py

```diff
@@ -211,6 +221,10 @@ def safe_summary(self, encoded):
         raise NotImplementedError('subclasses of BasePasswordHasher must provide a safe_summary() method')
 
     def must_update(self, encoded):
+        decoded = self.decode(encoded)
+        if 'salt' not in decoded:
+            warnings.warn('Decode not fully implemented. Decode will be required in future releases.',
```

Comment:
This warning text could include `type(self).__name__` to provide a bit of additional guidance.

Reply:
Updated and changed the associated test to match on a regex to fill in for `type(self).__name__`